### PR TITLE
Feature/client side date validation

### PR DESCRIPTION
**Pull request by StRhys**
### What

This adds client side validation as part of [this ticket](https://trello.com/c/T8PlqbQ3).

- Removed some instances of `Parallel()` for a few tests. Using `helper` was causing some date race issues.
- Fixed an issue with before date being before the after date and no error text showing.
- Added component tests for date fields.

### How to review

Sense check
or
Pull this branch
Run dp-frontend-search-controller using make debug ENABLE_REWORKED_DATA_AGGREGATION_PAGES=true
Grab latest dp design system and run it.
Port forward the api router to `sandbox dp ssh sandbox web 1 -p 23200:localhost:10800`
Head other to http://localhost:25000/datalist and play around with the dates.
Any invalidate date should be stopped before it gets into the url.

### Who can review

Frontend

### CODE REVIEW

**Comment by lindenmckenzie** [comment]
Tested:
- days greater than 31
- months greater than 12
- combinations of incorrect dates (31st September)
- leap years

Looks good.

**Review by lindenmckenzie — COMMENTED** [review]
Given the non-urgent nature of these changes, this might be a good opportunity to add some component tests for the validation.

File: features/datefields.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        Then element "#error-summary-title" should not be visible
+
+    Scenario Outline: Submitting the form with invalid date fields
+        Given I fill in input element "#after-date-day" with value "<day>"
+        And I fill in input element "#after-date-month" with value "<month>"
+        And I fill in input element "#after-date-year" with value "<year>"
+        When I click the "#search-filter > button" element
+        And I wait 1 seconds
+        Then element "#error-summary-title" should be visible
+        And element "#fromDate-error" should be visible
```
Comment (lindenmckenzie) [inline features/datefields.feature:26]: I don't *think* you need the wait here as 'should be visible' has a built in wait, might be wrong though.

File: features/datefields.feature [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    Scenario: Submitting the form with valid date fields
+        Given I fill in input element "#after-date-day" with value "01"
+        And I fill in input element "#after-date-month" with value "01"
+        And I fill in input element "#after-date-year" with value "2022"
+        And I fill in input element "#before-date-day" with value "31"
+        And I fill in input element "#before-date-month" with value "12"
+        And I fill in input element "#before-date-year" with value "2022"
+        When I click the "#search-filter > button" element
+        And I wait 1 seconds
+        Then element "#error-summary-title" should not be visible
```
Comment (lindenmckenzie) [inline features/datefields.feature:17]: Is there something better to wait for here?

File: features/datefields.feature [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    Scenario Outline: Submitting the form with invalid date fields
```
Comment (lindenmckenzie) [inline features/datefields.feature:19]:
```suggestion
    Scenario Outline: Submitting the form with invalid after-date fields
```

File: features/datefields.feature [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        Examples:
+            | day | month | year |
+            | 32  | 01    | 2022 |
+            | 01  | 13    | 2022 |
+            | 01  | 01    | 0000 |
+            | 31  | 09    | 2022 |
+            | 01  | 01    |      |
+
+    Scenario: Submitting the form with after date that's after before date
```
Comment (lindenmckenzie) [inline features/datefields.feature:53]:
```suggestion
    Scenario: Submitting the form with after-date that's after before-date
```
Possibly improves clarity?

**Review by lindenmckenzie — APPROVED** [review]
Have included some things to think about but happy to move it forward as is if you don't want to look into them.

**Review by mr-nick17 — APPROVED** [review]

**Pull request merged**